File: AIPO/C++/2016-Prelim-p6.cpp
#include <iostream>
#include <stdio.h>
#include <math.h>
#include <string>

using namespace std;

unsigned long long int modPow = pow(10,9) + 7;

unsigned long long int fac(int n){
	unsigned long long int res = 1;
	for(int i = 2; i <= n; i++){
		res=(i * res);
	}
	return res;
}

void solve(int n, int k){
	unsigned long long int top = 1;
	for(int i = n-k+1; i <=n; i++){
		top=(top*i);
	}
	top = top;
	unsigned long long int bottom = fac(k);
	unsigned long long int result = (top/bottom) % modPow;
	cout << top << "/" << bottom << endl;
	cout<<result<<endl;
}

int main(){
	int t,n,k;
	scanf("%i",&t);
	for(int i = 0; i < t; i++){
		scanf("%i %i",&n,&k);
		solve(n,k);
	}
	return 0;
}

File: AIPO/Java/2016-Prelim-p6.java
import java.util.*;
import java.lang.*;
import java.io.*;
import java.math.BigInteger;

class p6
{
	final static long mod_limit = (long) Math.pow(10,9) + 7;
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner scanner = new Scanner(new InputStreamReader(System.in));
		int t = scanner.nextInt();
		int n,k,result;
		BigInteger top,bottom,temp;
		for(int i = 0; i < t; i++){
			n=k=0;
			result=0;
			n=scanner.nextInt();
			k=scanner.nextInt();
			temp=top=bottom=BigInteger.ONE;
			for(int j=(n-k+1); j <= n; j++){
				top = top.multiply(BigInteger.valueOf(j));
			}
			for(int j=k; j>1; j--){
				bottom = bottom.multiply(BigInteger.valueOf(j));
			}
			temp = top.divide(bottom);
			temp = temp.mod(BigInteger.valueOf(mod_limit));
			System.out.println(temp.toString());
		}
	}
}